src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -1369,9 +1369,18 @@ private static bool IsCompilerAnalyzer(DiagnosticAnalyzer analyzer)
 
         public void Dispose()
         {
-            this.CompilationEventQueue?.TryComplete();
-            this.DiagnosticQueue?.TryComplete();
-            _queueRegistration.Dispose();
+            Dispose(true);
+            GC.SuppressFinalize(this);
```

Comment:
The implementation of `IDisposable.Dispose()` is sealed. Unsealed disposable types include a `GC.SuppressFinalize(this)` even if they do not directly own unmanaged resources to ensure correct behavior for a derived type which introduces a finalizer.